crates/metrics/src/txpool_metrics.rs

```diff
@@ -1,24 +1,97 @@
-use crate::global_registry;
-use prometheus_client::metrics::histogram::Histogram;
+use crate::{
+    buckets::{
+        buckets,
+        Buckets,
+    },
+    global_registry,
+};
+use prometheus_client::metrics::{
+    gauge::Gauge,
+    histogram::Histogram,
+};
 use std::sync::OnceLock;
 
 pub struct TxPoolMetrics {
-    pub tx_size_histogram: Histogram,
+    /// Size of transactions in the txpool in bytes
+    pub tx_size: Histogram,
+    /// Number of transactions in the txpool
+    pub number_of_transactions: Gauge,
+    /// Number of transactions pending verification, before being inserted in the txpool
+    pub number_of_transactions_pending_verification: Gauge,
+    /// Number of transactions that can be included in the next block
+    pub number_of_executable_transactions: Gauge,
+    /// Time of transactions in the txpool in seconds
+    pub transaction_time_in_txpool_secs: Histogram,
+    /// Time actively spent by transaction insertion in the thread pool
+    pub transaction_insertion_time_in_thread_pool_milliseconds: Histogram,
+    /// How long it took for the selection algorithm to select transactions
+    pub select_transaction_time_nanoseconds: Histogram,
 }
 
 impl Default for TxPoolMetrics {
     fn default() -> Self {
-        let tx_sizes = Vec::new();
+        let tx_size = Histogram::new(buckets(Buckets::TransactionSize));
+        let transaction_time_in_txpool_secs = Histogram::new(buckets(Buckets::Timing));
+        let select_transaction_time_nanoseconds =
```

Comment:
I think we need to rename it to be `select_transactions_time_nanoseconds`

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2385/commits/7c79307de926c4f2873c3c21e588faa3b11c49b2